src/main/java/gregtech/client/model/modelfactories/ActiveVariantBlockBakedModel.java

```diff
@@ -1,126 +1,118 @@
-        //Some mods like to call this without getting the extendedBlockState leading to a NPE crash since the
-        //unlisted ACTIVE property is null.
-        if (((IExtendedBlockState) state).getValue(VariantActiveBlock.ACTIVE) == null) {
-            activeState = false;
-        } else {
-            activeState = ((IExtendedBlockState) state).getValue(VariantActiveBlock.ACTIVE);
-        }
+        if (state == null) return Collections.emptyList();
 
-        ResourceLocation registryName = state.getBlock().getRegistryName();
-        if (registryName == null) {
-            throw new IllegalArgumentException("Block " + state.getBlock().getTranslationKey() + " has null registry name");
-        }
-        if (activeState) {
-            mrl = new ModelResourceLocation(registryName,
-                    "active=true,variant=" + state.getProperties().entrySet().stream().filter(p -> p.getKey().getName().equals("variant")).map(e -> {
-                        IProperty<?> p = e.getKey();
-                        return getPropertyName(p, e.getValue());
-                    }).collect(Collectors.joining()));
-        } else {
-            mrl = new ModelResourceLocation(registryName,
-                    "active=false,variant=" + state.getProperties().entrySet().stream().filter(p -> p.getKey().getName().equals("variant")).map(e -> {
-                        IProperty<?> p = e.getKey();
-                        return getPropertyName(p, e.getValue());
-                    }).collect(Collectors.joining()));
-        }
-        IBakedModel m = Minecraft.getMinecraft().blockRenderDispatcher.getBlockModelShapes().getModelManager().getModel(mrl);
-        TextureAtlasSprite textureAtlasSprite = m.getParticleTexture();
-        particle.set(textureAtlasSprite);
-        if (MinecraftForgeClient.getRenderLayer() == BloomEffectUtil.getRealBloomLayer()) {
-            if (ConfigHolder.client.casingsActiveEmissiveTextures) {
-                quads = new ArrayList<>();
-                for (BakedQuad b : m.getQuads(state, side, rand)) {
-                    if (b.getSprite().getIconName().contains("bloom")) {
-                        quads.add(b);
-                    }
-                }
-            }
-            else {
-                quads = Collections.emptyList();
-            }
+        IBakedModel m = getModel(state);
+
+        // If bloom is enabled, bloom textures are rendered on bloom layer
+        // If bloom is disabled (either by model specific bloom config or the presence of O**ifine shaders)
+        // it is rendered on CUTOUT layer instead.
+        if (getBloomConfig()) {
+            return MinecraftForgeClient.getRenderLayer() != BloomEffectUtil.BLOOM ?
+                    m.getQuads(state, side, rand) :
+                    m.getQuads(state, side, rand).stream()
+                            .filter(q -> q.getSprite().getIconName().contains("bloom"))
+                            .collect(Collectors.toList());
```

Comment:
To improve readability:

```suggestion
            return MinecraftForgeClient.getRenderLayer() == BloomEffectUtil.BLOOM ?
                    m.getQuads(state, side, rand).stream()
                            .filter(q -> q.getSprite().getIconName().contains("bloom"))
                            .collect(Collectors.toList()) :
                    m.getQuads(state, side, rand);
```